sympy/polys/rings.py

```diff
@@ -266,13 +266,10 @@ def __new__(cls, symbols, domain, order=lex):
 
     def _gens(self):
         """Return a list of polynomial generators. """
-        one = self.domain.one
         _gens = []
         for i in range(self.ngens):
-            expv = self.monomial_basis(i)
-            poly = self.zero
-            poly[expv] = one
-            _gens.append(poly)
+            generator = self._construct_generator(i)
+            _gens.append(generator)
         return tuple(_gens)
```

Comment:
I don't think we need to change this method because the Flint version can call `.gens()` on the flint ring.